### DISCUSSION TITLE
How to decompress the GZipStream returned by the function GetContentAsync(true)?

### DISCUSSION THREAD

**Discussion by sidediallo** [pnp/pnpcore#767] [Q&A]
Hello,

I'm trying to read large files (about 150 Mb) from SharePoint online.
I connect and load the file with no problem.
Then I use "GetContentAsync(true)" so I can iterate through all the contents of the file.
The problem is that when I do "await fileToDownload.GetContentAsync(true)" I get a GZipStream type instead of Stream.
and when I try to decompress the data obtained I get an exception "The archive entry was compressed using an unsupported compression method.".
How can I get decompressed data?
Thanks in advance. 

````
private async Task Read()
{
	try
	{
		using var context = await pnpContextFactory.CreateAsync("Default");
		
		string documentUrl = $"{context.Uri.PathAndQuery}/{filePath}/{fileName}";

		IFile fileToDownload = await context.Web.GetFileByServerRelativeUrlAsync(documentUrl);

		var downloadedContentStream = await fileToDownload.GetContentAsync(true);
		
		var bufferSize = 2 * 1024 * 1024;  // 2 MB

		var allBuffer = new List<byte[]>();

		var buffer = new byte[bufferSize];

		int read;

		while ((read = await downloadedContentStream.ReadAsync(buffer, 0, buffer.Length)) > 0)
		{
			allBuffer.Add(buffer.ToList().GetRange(0, read).ToArray());
		}

		var stream = new MemoryStream(allBuffer.SelectMany(o => o).ToArray());

                ...........
	}
	catch (Exception e)
	{
		Console.Error.Write(e.Message);
		
		throw;
	}
}
````

**Comment by jansenbe** [comment]
Hi @sidediallo , 

Below approach works fine. You seem to want to load all the returned data of the large file in memory? Not ideal for performance, especially when you need to deal with very large files. Probably better to first persist the file in temp location and then load it (or parts of it) and do the needed processing?

![image](https://user-images.githubusercontent.com/7451219/155278691-5ccbad64-2fbc-48de-aaca-8057c05045e6.png)
